arch/BPF/BPFMapping.c

```diff
@@ -35,99 +34,100 @@ const char *BPF_group_name(csh handle, unsigned int id)
+
+	{ BPF_INS_LE16, "le16" },	{ BPF_INS_LE32, "le32" },
+	{ BPF_INS_LE64, "le64" },	{ BPF_INS_BE16, "be16" },
+	{ BPF_INS_BE32, "be32" },	{ BPF_INS_BE64, "be64" },
+	{ BPF_INS_BSWAP16, "bswap16" }, { BPF_INS_BSWAP32, "bswap32" },
 	{ BPF_INS_BSWAP64, "bswap64" },
 
-	{ BPF_INS_LDW, "ldw" },
-	{ BPF_INS_LDH, "ldh" },
-	{ BPF_INS_LDB, "ldb" },
-	{ BPF_INS_LDDW,	"lddw" },
-	{ BPF_INS_LDXW, "ldxw" },
-	{ BPF_INS_LDXH, "ldxh" },
-	{ BPF_INS_LDXB, "ldxb" },
-	{ BPF_INS_LDXDW, "ldxdw" },
+	{ BPF_INS_LDW, "ldw" },		{ BPF_INS_LDH, "ldh" },
+	{ BPF_INS_LDB, "ldb" },		{ BPF_INS_LDDW, "lddw" },
+	{ BPF_INS_LDXW, "ldxw" },	{ BPF_INS_LDXH, "ldxh" },
+	{ BPF_INS_LDXB, "ldxb" },	{ BPF_INS_LDXDW, "ldxdw" },
+	{ BPF_INS_LDABSW, "ldabsw" },	{ BPF_INS_LDABSH, "ldabsh" },
+	{ BPF_INS_LDABSB, "ldabsb" },	{ BPF_INS_LDINDW, "ldindw" },
+	{ BPF_INS_LDINDH, "ldindh" },	{ BPF_INS_LDINDB, "ldindb" },
+
+	{ BPF_INS_STW, "stw" },		{ BPF_INS_STH, "sth" },
+	{ BPF_INS_STB, "stb" },		{ BPF_INS_STDW, "stdw" },
+	{ BPF_INS_STXW, "stxw" },	{ BPF_INS_STXH, "stxh" },
+	{ BPF_INS_STXB, "stxb" },	{ BPF_INS_STXDW, "stxdw" },
+	{ BPF_INS_XADDW, "xaddw" },	{ BPF_INS_XADDDW, "xadddw" },
+
+	{ BPF_INS_JA, "ja" },		{ BPF_INS_JEQ, "jeq" },
+	{ BPF_INS_JGT, "jgt" },		{ BPF_INS_JGE, "jge" },
+	{ BPF_INS_JSET, "jset" },	{ BPF_INS_JNE, "jne" },
+	{ BPF_INS_JSGT, "jsgt" },	{ BPF_INS_JSGE, "jsge" },
+	{ BPF_INS_CALL, "call" },	{ BPF_INS_CALLX, "callx" },
+	{ BPF_INS_EXIT, "exit" },	{ BPF_INS_JLT, "jlt" },
+	{ BPF_INS_JLE, "jle" },		{ BPF_INS_JSLT, "jslt" },
+	{ BPF_INS_JSLE, "jsle" },
+
+	{ BPF_INS_JAL, "jal" },		{ BPF_INS_JEQ32, "jeq32" },
+	{ BPF_INS_JGT32, "jgt32" },	{ BPF_INS_JGE32, "jge32" },
+	{ BPF_INS_JSET32, "jset32" },	{ BPF_INS_JNE32, "jne32" },
+	{ BPF_INS_JSGT32, "jsgt32" },	{ BPF_INS_JSGE32, "jsge32" },
+	{ BPF_INS_JLT32, "jlt32" },	{ BPF_INS_JLE32, "jle32" },
+	{ BPF_INS_JSLT32, "jslt32" },	{ BPF_INS_JSLE32, "jsle32" },
 
-	{ BPF_INS_STW, "stw" },
-	{ BPF_INS_STH, "sth" },
-	{ BPF_INS_STB, "stb" },
-	{ BPF_INS_STDW,	"stdw" },
-	{ BPF_INS_STXW, "stxw" },
-	{ BPF_INS_STXH, "stxh" },
-	{ BPF_INS_STXB, "stxb" },
-	{ BPF_INS_STXDW, "stxdw" },
-	{ BPF_INS_XADDW, "xaddw" },
-	{ BPF_INS_XADDDW, "xadddw" },
+	{ BPF_INS_RET, "ret" },
 
-	{ BPF_INS_JMP, "jmp" },
-	{ BPF_INS_JEQ, "jeq" },
-	{ BPF_INS_JGT, "jgt" },
-	{ BPF_INS_JGE, "jge" },
-	{ BPF_INS_JSET, "jset" },
-	{ BPF_INS_JNE, "jne" },
-	{ BPF_INS_JSGT,	"jsgt" },
-	{ BPF_INS_JSGE,	"jsge" },
-	{ BPF_INS_CALL,	"call" },
-	{ BPF_INS_CALLX, "callx" },
-	{ BPF_INS_EXIT,	"exit" },
-	{ BPF_INS_JLT, "jlt" },
-	{ BPF_INS_JLE, "jle" },
-	{ BPF_INS_JSLT, "jslt" },
-	{ BPF_INS_JSLE,	"jsle" },
+	{ BPF_INS_AADD, "aadd" },	{ BPF_INS_AOR, "aor" },
+	{ BPF_INS_AAND, "aand" },	{ BPF_INS_AXOR, "axor" },
+	{ BPF_INS_AFADD, "afadd" },	{ BPF_INS_AFOR, "afor" },
+	{ BPF_INS_AFAND, "afand" },	{ BPF_INS_AFXOR, "afxor" },
 
-	{ BPF_INS_RET, "ret" },
+	{ BPF_INS_AXCHG64, "axchg64" }, { BPF_INS_ACMPXCHG64, "acmpxchg64" },
```

Comment:
Please don't forget about the mnemonic thing. Either use the default mnemonics or add a flag to allow the user to switch between naming.

Reply:
I've addressed that in the last comment - I've checked the Linux kernel and this is the mnemonics they use. It's only GNU that use the "wrong" one.
I think this should be the default, but if you still think we should add flag I'll add.